src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -159,25 +158,40 @@ object BlockVTK {
     }
     offsetIterator(offsets, localOffsets.toSeq)
   }
+
+  /*
+   * FIXME: This is very inefficient and we probably want to use the right data
+   * structures to deal with duplicates for us.
+   */
+  private def distinctArrays(vertices: Seq[Array[Double]], previous: Seq[Array[Double]]=Nil): Seq[Array[Double]] = {
+    vertices match {
+      case Nil => previous
+      case vert+:verts =>
+        if (previous.exists(previousVertex => previousVertex sameElements vert)) {
+          distinctArrays(verts, previous)
+        } else {
+          distinctArrays(verts, vert+:previous)
+        }
+    }
+  }
 }
 
 /**
   * Simple data structure to contain data that represents a rock block that can be turned into vtk format
-  * @param Rock block 
   * @constructor Create a new rock block in format that can be turned into vtk by rockProcessor
   */
 case class BlockVTK(block: Block) {
   val orientedVertices = BlockVTK.orientVertices(block)
-  val tupleVertices = orientedVertices.values.flatten.toSeq.distinct
-  val connectivityMap = BlockVTK.connectivity(orientedVertices, tupleVertices)
-  val vertices = tupleVertices flatMap { t =>
-    List(t._1, t._2, t._3)
+  val arrayVertices = BlockVTK.distinctArrays(orientedVertices.values.flatten.toSeq)
```

Comment:
I think the `Double`s from the `WrappedArray` would just get garbage collected, but I also don't think this function would be too bad if the arrays are on average reasonably small.

Let's keep it as is for now, and we can do some profiling later to see what's really slowing us down. If this is actually slow, we can revisit it.